### PR TITLE
Move contextual and ast-viewer files into language-support

### PR DESCRIPTION
**Pull request by norascheuch**
This PR is part of an endeavour to order files in the codebase according to the domain model. Here the folder `src/contextual` is moved into `language-support`.

- some methods and classes inside the contextual folder were using `default` to export. These defaults were removed and an `index.ts` introduced
- some tests were moved to a new `language-support/contextual`- and `language-support-ast-viewer`-folder
- for 2 ast-viewer related data files a new folder and adjusted paths had to be added

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/language-support/index.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+export * from "./contextual";
```
Comment (charisk) [inline extensions/ql-vscode/src/language-support/index.ts:1]: Did you mean to export everything from `contextual` to the outside world? I would have thought that some of it is only internal to `language-support` (e.g. the `file-range-from-uri` stuff).

File: extensions/ql-vscode/src/language-support/contextual/index.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export * from "../ast-viewer/ast-builder";
```
Comment (charisk) [inline extensions/ql-vscode/src/language-support/contextual/index.ts:1]: Hmm this doesn't look right, why is it needed?

**Review by charisk — COMMENTED** [review]
Generally looking good, just some minor comments/questions.

File: extensions/ql-vscode/src/language-support/contextual/index.ts (hunk 2 above)
Comment (charisk) [reply extensions/ql-vscode/src/language-support/contextual/index.ts:1]: Come to think of it, while is there another index.ts file here? I was under the impression we were going to have 1 at each top-level directory (unless there were other reasons that make it difficult, like in a previous PR).

File: extensions/ql-vscode/src/language-support/index.ts (hunk 1 above)
Comment (norascheuch) [reply extensions/ql-vscode/src/language-support/index.ts]: Discussed synchronously, fixed

File: extensions/ql-vscode/src/common/commands.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import type { CommandManager } from "../packages/commands";
 import type { Uri, Range, TextDocumentShowOptions } from "vscode";
-import type { AstItem } from "../astViewer";
+import type { AstItem } from "../language-support/ast-viewer/ast-viewer";
```
Comment (charisk) [inline extensions/ql-vscode/src/common/commands.ts:3]: Is this missing from index.ts?

File: extensions/ql-vscode/src/extension.ts [hunk 4]
```diff
@@ -21,7 +21,7 @@ import { testExplorerExtensionId, TestHub } from "vscode-test-adapter-api";
 import { lt, parse } from "semver";
 import { watch } from "chokidar";
 
-import { AstViewer } from "./astViewer";
+import { AstViewer } from "./language-support/ast-viewer/ast-viewer";
```
Comment (charisk) [inline extensions/ql-vscode/src/extension.ts:24]: Is this missing from index.ts?

File: extensions/ql-vscode/src/extension.ts [hunk 5]
```diff
@@ -119,7 +117,7 @@ import {
   QueryServerCommands,
 } from "./common/commands";
 import { LocalQueries, QuickEvalCodeLensProvider } from "./local-queries";
-import { getAstCfgCommands } from "./ast-cfg-commands";
+import { getAstCfgCommands } from "./language-support/ast-viewer/ast-cfg-commands";
```
Comment (charisk) [inline extensions/ql-vscode/src/extension.ts:120]: Is this missing from index.ts?

**Review by charisk — APPROVED** [review]
Looks good, just some potential improvements.

**Pull request merged**